src/motor_node.cc

```diff
@@ -181,47 +185,55 @@ int main(int argc, char* argv[]) {
         firmware_params.option_switch = robot->getOptionSwitch();
         ROS_INFO_ONCE("Setting firmware option register to 0x%x.", firmware_params.option_switch);
         robot->setOptionSwitchReg(firmware_params.option_switch);
-        ctrlLoopDelay.sleep();        // Allow controller to process command
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
+    }
+    
+    if (robot->firmware_version >= MIN_FW_SYSTEM_EVENTS) {
+        // Start out with zero for system events
+        robot->setSystemEvents(0);  // Clear entire system events register
+        robot->system_events = 0;
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
     }
 
     // Setup other firmware parameters that could come from ROS parameters
     if (robot->firmware_version >= MIN_FW_ESTOP_SUPPORT) {
         robot->setEstopPidThreshold(firmware_params.estop_pid_threshold);
-        ctrlLoopDelay.sleep();        // Allow controller to process command
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
         robot->setEstopDetection(firmware_params.estop_detection);
-        ctrlLoopDelay.sleep();        // Allow controller to process command
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
     }
 
     if (robot->firmware_version >= MIN_FW_MAX_SPEED_AND_PWM) {
         robot->setMaxFwdSpeed(firmware_params.max_speed_fwd);
-        ctrlLoopDelay.sleep();        // Allow controller to process command
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
         robot->setMaxRevSpeed(firmware_params.max_speed_rev);
-        ctrlLoopDelay.sleep();        // Allow controller to process command
+        ros::Duration(mcbStatusPeriodSec).sleep();  // allow 2 status replies
     }
 
     ros::Time last_time;
     ros::Time current_time;
-    ros::Time last_sys_event_query_time;
     ros::Duration elapsed;
-    ros::Duration elapsed_since_sys_event_query;
-    ros::Duration sysEventQueryPeriod(15.0);     // A periodic query of MCB
+    ros::Time last_sys_maint_time;
+    ros::Duration elapsed_since_last_action;
```

Comment:
There is no point in declaring this way up here, we should just declare it where we use it.

Reply:
I'm mixed on this but did it.